### PR TITLE
test: Calculation logic test for subscription calculation

### PR DESCRIPTION
**Pull request by JesperGomes**
# Description

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002` or `sandbox_002`)
- [ ] Documentation has been updated
- [ ] The integration [event catalog](https://energinet.atlassian.net/wiki/spaces/D3/pages/555581556/Event+catalog) has been updated
- [ ] C4 diagrams have been updated and Team Outlaws has been informed about relevant changes

### CODE REVIEW

File: source/databricks/calculation_engine/tests/helpers/data_frame_utils.py [hunk 1]
```diff
@@ -44,7 +43,7 @@ def assert_dataframe_and_schema(
     ignore_decimal_precision: bool = False,
     columns_to_skip: any = None,
 ) -> None:
-
+    show_dataframes(actual, expected)
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/helpers/data_frame_utils.py:46]: remove

File: source/databricks/calculation_engine/tests/calculation_logic/conftest.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 import pytest
 from pyspark.sql import SparkSession
 
-from calculation_logic.scenario_factory import ScenarioFixture
+from tests.calculation_logic.scenario_factory import ScenarioFixture
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/calculation_logic/conftest.py:17]: remove

File: source/databricks/calculation_engine/tests/calculation_logic/features/wholesale_calculations/subscription_amount_per_charge/when_charge_link_starts_later/test_data/time_series_points.csv [hunk 3]
```diff
@@ -0,0 +1 @@
+﻿metering_point_id;quantity;quality;observation_time
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/calculation_logic/features/wholesale_calculations/subscription_amount_per_charge/when_charge_link_starts_later/test_data/time_series_points.csv:1]: delete file

File: source/databricks/calculation_engine/tests/calculation_logic/features/wholesale_calculations/wholesale_results_dataframe.py [hunk 4]
```diff
@@ -76,5 +76,6 @@ def create_wholesale_result_dataframe(*args) -> DataFrame:  # type: ignore
             ",",
         ).cast(ArrayType(StringType())),
     )
+    df = df.withColumn(WholesaleResultColumnNames.calculation_result_id, lit(""))
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/calculation_logic/features/wholesale_calculations/wholesale_results_dataframe.py:79]: delete?

File: source/databricks/calculation_engine/package/calculation/output/schemas/wholesale_results_schema.py [hunk 5]
```diff
@@ -45,7 +45,7 @@
         StructField(
             WholesaleResultColumnNames.quantity_qualities,
             ArrayType(StringType()),
-            False,
+            True,
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/output/schemas/wholesale_results_schema.py:48]: needed?

Comment (JesperGomes) [reply source/databricks/calculation_engine/package/calculation/output/schemas/wholesale_results_schema.py:48]: yes

**Review by energinet-ajw — APPROVED** [review]

**Pull request merged**